src/mscorlib/src/System/Span.cs

```diff
@@ -625,5 +636,266 @@ internal static unsafe void CopyTo<T>(ref T destination, ref T source, int eleme
+                        return;
+                    case 12:
+                        *(long*)bytePointer = (long)0;
+                        *(int*)(bytePointer + 8) = (int)0;
+                        return;
+                    case 13:
+                        *(long*)bytePointer = (long)0;
+                        *(int*)(bytePointer + 8) = (int)0;
+                        *(bytePointer + 12) = (byte)0;
+                        return;
+                    case 14:
+                        *(long*)bytePointer = (long)0;
+                        *(int*)(bytePointer + 8) = (int)0;
+                        *(short*)(bytePointer + 12) = (short)0;
+                        return;
+                    case 15:
+                        *(long*)bytePointer = (long)0;
+                        *(int*)(bytePointer + 8) = (int)0;
+                        *(short*)(bytePointer + 12) = (short)0;
+                        *(bytePointer + 14) = (byte)0;
+                        return;
+                    case 16:
+                        *(long*)bytePointer = (long)0;
+                        *(long*)(bytePointer + 8) = (long)0;
+                        return;
+                    case 17:
+                        *(long*)bytePointer = (long)0;
+                        *(long*)(bytePointer + 8) = (long)0;
+                        *(bytePointer + 16) = (byte)0;
+                        return;
+                    case 18:
+                        *(long*)bytePointer = (long)0;
+                        *(long*)(bytePointer + 8) = (long)0;
+                        *(short*)(bytePointer + 16) = (short)0;
+                        return;
+                    case 19:
+                        *(long*)bytePointer = (long)0;
+                        *(long*)(bytePointer + 8) = (long)0;
+                        *(short*)(bytePointer + 16) = (short)0;
+                        *(bytePointer + 18) = (byte)0;
+                        return;
+                    case 20:
+                        *(long*)bytePointer = (long)0;
+                        *(long*)(bytePointer + 8) = (long)0;
+                        *(int*)(bytePointer + 16) = (int)0;
+                        return;
+                    case 21:
+                        *(long*)bytePointer = (long)0;
+                        *(long*)(bytePointer + 8) = (long)0;
+                        *(int*)(bytePointer + 16) = (int)0;
+                        *(bytePointer + 20) = (byte)0;
+                        return;
+                    case 22:
+                        *(long*)bytePointer = (long)0;
+                        *(long*)(bytePointer + 8) = (long)0;
+                        *(int*)(bytePointer + 16) = (int)0;
+                        *(short*)(bytePointer + 20) = (short)0;
+                        return;
+                }
+
+                // P/Invoke into the native version for large lengths
+                if (byteLength >= 512) goto PInvoke;
```

Comment:
I made the following change and performance degrades severely for when byteLength >= 512.
```C#
if (byteLength >= 512) goto PInvoke;  // From
if (byteLength >= 512) RuntimeImports.RhZeroMemory(ref b, byteLength); // To
```

![image](https://cloud.githubusercontent.com/assets/6527137/22960939/d48a7fa4-f2f6-11e6-95ac-c2a2b577b6ce.png)